### ISSUE TITLE
Strategy for working with images for different screen densities?

### ISSUE THREAD

**Issue by emiloberg** [NativeScript/NativeScript#203]
Is there a preferred way for working with images, more especially when it comes to serving images of different density/resolution.

My initial attempt was a simple `<Image src="res://myimage">` as it is done in the [TelerikNEXT example](https://github.com/NativeScript/sample-TelerikNEXT/blob/master/app/session-page.xml#L9).

However, that does not work with custom images. Requesting one of the out-of-the-box images such as `<Image src="res://Default" />` works perfectly, requesting a custom image such as `<Image src="res://myimage">` does _not_, it just turns out blank. Is it supposed to work? And if so, is that the preferred way of working with images? If not, what is?

Footnote: the `myimage` lives in `/App_Resources/iOS/` and there are 3 images; `myimage.png`, `myimage@2x.png` and `myimage@3x.png`. When inspecting `platforms/ios/rekapp/Resources/icons` all three images show up there.

**Comment by emiloberg** [comment]
Currently, I'm solving it with a little help from an image helper function which pick the the correct pic. Tthe path juggling is because I want to keep the paths give to me by [Google Material Design Icons](https://github.com/google/material-design-icons))

```js
import {fromFile as imageFromFile} from 'image-source';
import {screen} from 'platform';
import {android, ios} from 'application';

let scaleFactor = screen.mainScreen.scale;

function getImage(category, name) {
	let scalePath;
	let imagePath;
	if (ios) {
		imagePath = '~/images/md/' + category + '/ios/' + name + '.imageset/' + name;
		if (scaleFactor >= 3) {
			scalePath = '_3x';
		} else if (scaleFactor >= 2) {
			scalePath = '_2x';
		}
		imagePath = imagePath + scalePath + '.png';
	} else if (android) {
		// todo load for android
	}
	return imageFromFile(imagePath);
}

const Images = {
	left: getImage('navigation', 'ic_chevron_left'),
	menu: getImage('navigation', 'ic_menu'),
	search: getImage('action', 'ic_search')
};

export default Images;
```

**Comment by jlooper** [comment]
I believe that you need to have your images in the standard 3 sizes for iOS in the appname/platforms/ios/appname/Images.xcassets folders (each image set has its own folder). For Android, you are putting your images in appname/platforms/android/res/drawable-hdpi etc, without any specification of 2x, 3x, etc. 

I'd like to see the build allow us to put our scaled images into one place, and during the build process you'd have them be placed in the proper platform-specific folder. This way, when we use the markup res://xxx for the image, the content-scaled images are properly referenced. Is this something we could get working, or am I missing something?

**Comment by vakrilov** [comment]
Hi,
Adding images to the App_Resources and then using them in the app as resources is currently not working. It requires some additional modifications in the NativeScrtipt CLI(basically, the images should be included in the XCode project as well to be embedded in the app). There is an [issue for that](https://github.com/NativeScript/nativescript-cli/issues/520) and it will probably be fixed soon. 
Just for clarification - the images that are in the `/App_Resources/iOS/`in the default app template are already included in the xCode porject and that's why they work.

**Comment by N3ll** [comment]
Fixed in version 1.3.0.
